src/particle.cpp

```diff
@@ -36,6 +36,30 @@
 
 namespace openmc {
 
+double Particle::speed() const
+{
+  // Determine mass in eV/c^2
+  double mass;
+  switch (this->type()) {
+  case ParticleType::neutron:
+    mass = MASS_NEUTRON_EV;
+    break;
+  case ParticleType::photon:
+    mass = 0.0;
+    break;
+  case ParticleType::electron:
+  case ParticleType::positron:
+    mass = MASS_ELECTRON_EV;
+    break;
+  }
+
+  // Calculate inverse of Lorentz factor
+  const double inv_gamma = mass / (this->E() + mass);
```

Comment:
Pure formality, I would define gamma normally here (i.e. not inverse) and write a bit more to make it clear faster.

Reply:
I did it this way so that the same formula works for massless particles (photons). If you do it the "normal" way, it would result in a divide-by-zero for photons.

Reply:
yes, makes sense